src/util/feefrac.h

```diff
@@ -53,10 +57,19 @@ struct FeeFrac
 #ifdef __SIZEOF_INT128__
     static inline __int128 Mul(int64_t a, int32_t b) noexcept
     {
-        // If __int128 is available, use 128-bit wide multiply.
+        // If __int128 is available, use 128-bit wide multiplication.
         return __int128{a} * b;
     }
+#elif defined(_MSC_VER) && defined(_M_X64)
+    static inline std::pair<int64_t, uint64_t> Mul(int64_t a, int32_t b) noexcept
+    {
+        // On 64-bit MSVC, use _mul128 intrinsic for wide multiplication.
+        std::pair<int64_t, uint64_t> ret;
+        ret.second = _mul128(a, b, &ret.first);
+        return ret;
```

Comment:
> on MSVC one needs to use the `_mul128` or `_mulh` intrinsics instead

From my research, it follows that [`__mulh`](https://learn.microsoft.com/en-us/cpp/intrinsics/mulh) seems more performant.

This code:
```suggestion
        // On 64-bit MSVC, use __mulh intrinsic for wide multiplication.
        return {__mulh(a, b), std::bit_cast<uint64_t>(a) * b};
```
gives the following numbers for the [benchmark](https://github.com/bitcoin/bitcoin/pull/29758#issuecomment-2147006298):
```
> build_msvc\x64\Release\bench_bitcoin.exe -filter=FeefracMultipication

|               ns/op |                op/s |    err% |     total | benchmark
|--------------------:|--------------------:|--------:|----------:|:----------
|                1.41 |      707,119,862.93 |    2.6% |      0.17 | `FeefracMultipication`
> build_msvc\x64\Release\bench_bitcoin.exe -filter=FeefracMultipication

|               ns/op |                op/s |    err% |     total | benchmark
|--------------------:|--------------------:|--------:|----------:|:----------
|                1.36 |      734,209,621.15 |    0.9% |      0.16 | `FeefracMultipication`
> build_msvc\x64\Release\bench_bitcoin.exe -filter=FeefracMultipication

|               ns/op |                op/s |    err% |     total | benchmark
|--------------------:|--------------------:|--------:|----------:|:----------
|                1.39 |      720,775,228.66 |    3.1% |      0.17 | `FeefracMultipication`
```